tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/paging/PagingSendTest.java

```diff
@@ -288,6 +295,290 @@ public void testPagingDoesNotDuplicateBatchMessagesAfterPagingStarted() throws E
+            fail("should be immediate blocked on paging");
+         } catch (ActiveMQAddressFullException ex) {
+            //ok
+         }
+
+         assertTrue(queuePagingStore.isPageFull());
+
+         // set pageLimitBytes to bigger value to unblock paging again
+         addressSettings.setPageLimitBytes(Long.valueOf(size * 2));
+         server.getAddressSettingsRepository().addMatch(addressName, addressSettings);
+
+         // now page is enabled again
+         assertFalse(queuePagingStore.isPageFull());
+
+         sendMessageBatch(1, messageSize, session, queueAddr);
+         assertTrue(waitForMessages(queue, totalMessages + 1, 10000));
+      }
+   }
+
+   @Test
+   public void testPageLimitBytesAndPageLimitMessagesValidation() throws Exception {
+
+      final String queueName = getTestMethodName();
+
+      try (ClientSessionFactory sf = createSessionFactory(locator)) {
+         ClientSession session = sf.createSession(true, true);
+
+         SimpleString queueAddr = SimpleString.of(queueName);
+         session.createQueue(QueueConfiguration.of(queueAddr));
+
+         final int size = 1024 * 50;
+         final Long maxMessages = 10L;
+         AddressSettings addressSettings = new AddressSettings();
+         addressSettings.setAddressFullMessagePolicy(AddressFullMessagePolicy.PAGE);
+         addressSettings.setPageFullMessagePolicy(PageFullMessagePolicy.FAIL);
+         addressSettings.setPageSizeBytes(size);
+         addressSettings.setPageLimitBytes(Long.valueOf(size * 10));
+         addressSettings.setMaxSizeBytes(size);
+         addressSettings.setPageLimitMessages(maxMessages);
+
+         server.getAddressSettingsRepository().addMatch(queueName, addressSettings);
+
+         int totalMessages = 0;
+         int messageSize = 1024;
+         ClientProducer producer = session.createProducer(queueAddr);
+         boolean stop = false;
+         while (!stop) {
+            try {
+               ClientMessage message = createMessage(session, messageSize, totalMessages, null);
+               producer.send(message);
+               totalMessages++;
+            } catch (ActiveMQAddressFullException ex) {
+               stop = true;
+            }
+         }
+
+         Queue queue = server.locateQueue(queueAddr);
+         assertTrue(waitForMessages(queue, totalMessages, 10000));
+
+         PagingStore queuePagingStore = queue.getPagingStore();
+         assertTrue(queuePagingStore != null && queuePagingStore.isPageFull());
+
+         // now pages still under limit
+         long existingPages = queuePagingStore.getNumberOfPages();
+         assertTrue(existingPages <= queuePagingStore.getPageLimitBytes() / queuePagingStore.getPageSizeBytes());
+
+         // but the messages reach the limit
+         PageCursorProviderImpl cursorProvider = (PageCursorProviderImpl) queuePagingStore.getCursorProvider();
+         assertTrue(cursorProvider.getNumberOfMessagesOnSubscriptions() == maxMessages);
```

Comment:
Expand the use of PageCursorProviderTestAccessor rather than making the method public so you can do this